### ISSUE TITLE
Why does my batchnormal report an error here 

### ISSUE THREAD

**Issue by wangyaxin1998** [SciSharp/TensorFlow.NET#796]
![image](https://user-images.githubusercontent.com/81554655/114264375-23c87180-9a1d-11eb-9430-854b8dd0a7a1.png)
I use the old version of tensorflow.net to find a place and it will not report an error

**Comment by wangyaxin1998** [comment]
Because there is no tf.layers.batchnoraml method in tensorflow.net, I have to use batchnormal in keras. I also saw the example given by the author, layers.batchnormal can run through, but I don’t know what I am wrong where

**Comment by Oceania2018** [comment]
@wangyaxin1998 Can you provide runnable UnitTest?

**Comment by wangyaxin1998** [comment]
> @wangyaxin1998 Can you provide runnable UnitTest?

     public Graph BuildGraph()
        {

            var graph = new Graph().as_default();

            tf_with(tf.name_scope("define_input"), delegate
            {
                **input_data = tf.placeholder(dtype: tf.float32, shape: (-1, 416, 416, -1), name: "input_data");****
                label_sbbox = tf.placeholder(dtype: tf.float32, shape: (-1, 52, 52, -1, -1), name: "label_sbbox");
                label_mbbox = tf.placeholder(dtype: tf.float32, (-1, 26, 26, -1, -1), name: "label_mbbox");
                label_lbbox = tf.placeholder(dtype: tf.float32, (-1, 13, 13, -1, -1), name: "label_lbbox");

                true_sbboxes = tf.placeholder(dtype: tf.float32, (-1, -1, -1), name: "sbboxes");
                true_mbboxes = tf.placeholder(dtype: tf.float32, (-1, -1, -1), name: "mbboxes");
                true_lbboxes = tf.placeholder(dtype: tf.float32, (-1, -1, -1), name: "lbboxes");
                trainable = tf.placeholder(dtype: tf.float32, name: "training");
            });

            tf_with(tf.name_scope("define_loss"), delegate
            {
                **model = new YOLOV3(input_data, true);**
                net_var = tf.global_variables();
                (giou_loss, conf_loss, prob_loss) = model.compute_loss(label_sbbox, label_mbbox, label_lbbox, true_sbboxes, true_mbboxes, true_lbboxes);
                loss = giou_loss + conf_loss + prob_loss;
            });

            public YOLOV3(Tensor input_data, bool trainable)
            **(conv_lbbox , conv_mbbox , conv_sbbox)= _build_nework(input_data);**


   public (Tensor, Tensor, Tensor) _build_nework(Tensor input_data)
        {
            Tensor route_1, route_2;
            **(route_1, route_2, input_data) = backbone.darknet53(input_data, trainable);**

                tf_with(tf.variable_scope("darknet"), delegate
            {
                **input_data = Common.convolutional(input_data, filters_shape: new int[] { 3, 3, 3, 32 }, trainable: trainable, name: "conv0");**


 public static Tensor convolutional(Tensor input_data, int[] filters_shape, bool trainable, string name, bool downsample = false, bool activate = true, bool bn = true)
        {
            Tensor conv = null;
            tf_with(tf.variable_scope(name), delegate
            {
                string padding = "SAME";
                var strides = new int[] { 1, 1, 1, 1 };
                //var strides = 1;
                if (downsample)
                {                    padding = "VALID";
                    strides = new int[] { 1, 2, 2, 1 };
                    //strides = 2;
                    var nd = np.array(new int[,] { { 1, 0 }, { 1, 0 } });
                    input_data = keras.layers.ZeroPadding2D(nd).Apply((input_data));
                }

                var weight = tf.compat.v1.get_variable(name: "weight", dtype: tf.float32, trainable: true, shape: filters_shape, initializer:tf.random_normal_initializer(stddev: 0.01f));//, initializer: tf.random_normal_initializer(stddev: 0.01f)
                conv = tf.nn.conv2d(input: input_data, filter: weight, strides: strides, padding: padding);
                IVariableV1 bias;
                if (bn)
                {
                    **conv = keras.layers.BatchNormalization().Apply(conv);**   
                }
                else
                {
                    bias = tf.compat.v1.get_variable(name: "bias", shape: filters_shape[filters_shape.Length - 1], trainable: true, dtype: tf.float32, initializer: tf.constant_initializer(0.0f));
                    conv = tf.nn.bias_add(conv, bias);
                }
                if (activate)
                    conv = tf.nn.leaky_relu(conv, alpha: 0.1f);
            });
            return conv;
        }

**Comment by wangyaxin1998** [comment]
> @wangyaxin1998 Can you provide runnable UnitTest?

I am very sorry that I took your time, I completely imitated the C# version of the code written by https://github.com/YunYang1994/tensorflow-yolov3

**Comment by Oceania2018** [comment]
@wangyaxin1998 You should use [v2 version](https://github.com/YunYang1994/TensorFlow2.0-Examples/tree/master/4-Object_Detection/YOLOV3).

**Comment by wangyaxin1998** [comment]
> @wangyaxin1998 You should use [v2 version](https://github.com/YunYang1994/TensorFlow2.0-Examples/tree/master/4-Object_Detection/YOLOV3).

![image](https://user-images.githubusercontent.com/81554655/114483184-86e31f80-9c3a-11eb-8ad8-f3fc959028fe.png)
![image](https://user-images.githubusercontent.com/81554655/114483628-48019980-9c3b-11eb-8806-715b1f92b9a9.png)

Thank you for your answer. I have implemented it with the version of tensorflow2.0 and found that when constructing the network, tf.concat cannot be used when the batch_size of the first dimension is none, and the merged dimension of the two tensors is unknown. When KerasApi.keras.layers.Concatenate().Apply can be used, it will affect the naming of certain layers of the network, such as tf_op_layer_concat (TensorFlowO [(None, 26, 26, 768) 0 and concatenate(Concatenate) (None , 26, 26, 768) 0
